azure_batch/main.tf

```diff
@@ -0,0 +1,183 @@
+
+# Notes
+
+1. Create a storage account [3] to store files used by the compute nodes to perform any backend 
+   calculation.
+2. Create a container within the above-mentioned storage account with a user-tunable retention 
+   policy of n days [4].
+3. Create a private container registry [5] and push a private Docker image in that registry.
+4. Create the Batch account. Warning: it is not possible to create more then a single Batch Account
+   per subscription and per region due to quota limitations (see Ref. [6]). Consider increasing that
+   if needed.
+5. Create a resizable pool (see Ref. [7] for more details about the autoscale formula) made of 
+   Docker-compatible nodes whose configuration is user-chosen [8]. Warning: by default, only a 
+   very limited array of VMs is available (10 VMs chosen amonst the Av2 Series, DSv3 Series, 
+   Dv3 Series, ESv3 Series and Ev3 Series). Should you need other types of VMs or more of them, 
+   consider asking for an increase of quota for your Azure subscription (see Ref. [6]).
+
+# Usage
+
+# Create a resource group that gathers all needed resources 
+resource "azurerm_resource_group" "batch_computing_rg" {
+  name     = "rg-batch-computing-${terraform.workspace}"
+  location = "West Europe"
+}
+
+module "azure_batch_service" {
+  source = "github.com/radix-ai/terraform-modules//azure_batch"
+  subscription_id              = "yoursubscriptionID"
+  resource_group_name          = azurerm_resource_group.batch_computing_rg.name
+  resource_group_location      = azurerm_resource_group.batch_computing_rg.location
+  container_name               = "inputcontainer"
+  keep_inp_files_during_n_days = 1
+  vm_size                      = "Standard_A1_V2"
+  image_name                   = "python"
+  image_tag                    = "3.8"
+  command_line                 = "echo 'Hello world'"
+  autoscale_formula            = <<EOF
+          $curTime = time();
+          $workHours = $curTime.hour >= 8 && $curTime.hour < 18;
+          $isWeekday = $curTime.weekday >= 1 && $curTime.weekday <= 5;
+          $isWorkingWeekdayHour = $workHours && $isWeekday;
+          $TargetDedicatedNodes = $isWorkingWeekdayHour ? 2:1;
+          $NodeDeallocationOption = taskcompletion;
+  EOF
+}
+
+# References
+
+[1] https://docs.microsoft.com/en-gb/azure/batch/
+[2] https://docs.microsoft.com/en-us/azure/batch/nodes-and-pools
+[3] https://docs.microsoft.com/en-us/azure/storage/common/storage-account-overview
+[4] https://docs.microsoft.com/en-us/azure/storage/blobs/immutable-time-based-retention-policy-overview
+[5] https://docs.microsoft.com/en-us/azure/container-registry/
+[6] https://docs.microsoft.com/en-us/azure/batch/batch-quota-limit
+[7] https://docs.microsoft.com/en-us/azure/batch/batch-automatic-scaling
+[8] https://docs.microsoft.com/en-us/azure/batch/batch-docker-container-workloads
+
+*/
+
+# Create an Azure Blob Storage linked to Azure Batch
```

Comment:
Would add this for extra information on why the storage account is linked with Azure Batch:
```suggestion
# Create an storage account linked to Azure Batch
# See: https://docs.microsoft.com/en-us/azure/batch/accounts
```